packages/kit/src/runtime/client/router.js

```diff
@@ -155,8 +184,6 @@ export class Router {
 			// Ignore if <a> has a target
 			if (a instanceof SVGAElement ? a.target.baseVal : a.target) return;
 
-			if (!this.owns(url)) return;
-
```

Comment:
IIUC, this is removed so in `_navigate` we can execute `before_navigate` callbacks? I'm not sure if this is right since if we navigate to external links, `onbeforeunload` should be called too? (which also calls` before_navigate`)

Perhaps I'm missing another piece of this.

Reply:
If on `beforeunload` is called before external navigation, then I would also tweak the docs and change "This includes navigation to external sites. `to` will be `null` if the user is closing the page." to "`to` will be `null` if the user is closing the page or navigating to an external site."

Reply:
Ah my bad, I just checked that `onbeforeunload` isn't called for external navigation, so I guess this is fine. I thought `before_navigate` would be called twice for external navigation.